### DISCUSSION TITLE
Bring back standardnote/web docker image for self hosting

### DISCUSSION THREAD

**Discussion by jackyzy823** [standardnotes/forum#2853] [Suggestions]
Since the standardnotes/app#1360 PR moves web-server to static web. so the docker image `standardnotes/web` will no longer receive update?

I know using AWS CloudFront + S3 to serve static content helps reducing the cost of web server. However it will be harder to self host `web` , no docker image anymore and the env variable can not be applied anymore.

Now as a self hosting user, You have to do all the yarn compilation stuff, **edit URL in index.html** (the worst part), and setup a server (like nginx/apache) or bundle static with server into a docker image to serve these static contents.

So could you make a docker image which serves static content and supports env variable? 

Thanks.

**Comment by jackyzy823** [comment]
Here I provide my solution for those who want to self host standardnotes/web: https://github.com/jackyzy823/standardnotes-web-image/ and prebuilt docker image https://github.com/jackyzy823/standardnotes-web-image/pkgs/container/standardnotes-web

**Reply by jpbaril** [reply]
Thank you so much!
I was so confused as to why my own self-hosted webapp was looking different than the one at app.standardnotes.com...

Question:
I see there are some environment variables that need to be set that I had not in my old version.
What should I put for "WEBSOCKET_URL" ?
For "ENABLE_UNFINISHED_FEATURES" I suppose the value should be "true" ?

Thanks again !

**Reply by jackyzy823** [reply]
> Question: I see there are some environment variables that need to be set that I had not in my old version. What should I put for "WEBSOCKET_URL" ? For "ENABLE_UNFINISHED_FEATURES" I suppose the value should be "true" ?


WEBSOCKET_URL empty is ok. This is used for notifying subscription update, which is not necessary for self hosting.
ENABLE_UNFINISHED_FEATURES empty is ok.  https://github.com/standardnotes/app/blob/main/packages/mobile/html/Web.bundle/src/index.html#L13 and it is not used anywhere in the codebase.

**Comment by sequent5** [comment]
> Here I provide my solution for those who want to self host standardnotes/web: https://github.com/jackyzy823/standardnotes-web-image/ and prebuilt docker image https://github.com/jackyzy823/standardnotes-web-image/pkgs/container/standardnotes-web

you are great!
